WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift

```diff
@@ -3407,3 +3420,27 @@ extension AztecPostViewController: PostEditorNavigationBarManagerDelegate {
         navigationItem.leftBarButtonItems = items
     }
 }
+
+// MARK: - Media Editing
+//
+extension AztecPostViewController {
+    func edit(_ imageAttachment: ImageAttachment) {
+        guard let image = imageAttachment.image else {
+            return
+        }
+
+        self.mediaEditor.edit(image, from: self,
+                  onFinishEditing: { image in
+                                if let image = image {
+                                    self.replace(attachment: imageAttachment, with: image)
+                                    self.mediaEditor.dismiss(animated: true)
+                                }
+        })
+    }
+
+    private func replace(attachment: ImageAttachment, with image: UIImage) {
+        let info = MediaAnalyticsInfo(origin: .editor(.deviceLibrary), selectionMethod: mediaSelectionMethod)
```

Comment:
Should we pass the MediaSource to the replace function instead of hardcoding to .deviceLibrary here?

Reply:
This is a good question. For some reason, I forgot to update that.

So, since this image is already presented in the post, I can't tell from where it comes. But we know that if it's being inserted again, it came from the Media Editor. I changed to `.editor(.mediaEditor)`. :)